sys/auto_init/saul/auto_init_pulse_counter.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/*
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization for PULSE_COUNTER devices
+ *
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeley.edu>
+ *
+ * @}
+ */
+
+#ifdef MODULE_PULSE_COUNTER
+
+#include "log.h"
+#include "saul_reg.h"
+#include "pulse_counter_params.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define PULSE_COUNTER_NUM    (sizeof(pulse_counter_params)/sizeof(pulse_counter_params[0]))
+
+/**
+ * @brief   Pulse count value
+ */
+static int16_t pulse_count[PULSE_COUNTER_NUM];
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+static pulse_counter_t pulse_counter_devs[PULSE_COUNTER_NUM];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[PULSE_COUNTER_NUM];
+
+/**
+ * @brief   Reference the driver struct
+ * @{
+ */
+extern saul_driver_t pulse_counter_saul_driver;
+/** @} */
+
+
+void auto_init_pulse_counter(void)
+{
+    for (unsigned i = 0; i < PULSE_COUNTER_NUM; i++) {
+        LOG_DEBUG("[auto_init_saul] initializing pulse_counter #%u\n", i);
+
+        pulse_counter_devs[i].pulse_count = &pulse_count[i];
```

Comment:
as stated above: why not save the counter value directly in the device descriptor?

Reply:
The answer is above :)